tests/src/unit-json_patch.cpp

```diff
@@ -60,14 +60,20 @@ TEST_CASE("JSON patch")
             json const doc2 = R"({ "q": { "bar": 2 } })"_json;
 
             // because "a" does not exist.
+#if JSON_DIAGNOSTIC_POSITIONS
+            CHECK_THROWS_WITH_AS(doc2.patch(patch1), "[json.exception.out_of_range.403] (bytes 0-21) key 'a' not found", json::out_of_range&);
+#else
             CHECK_THROWS_WITH_AS(doc2.patch(patch1), "[json.exception.out_of_range.403] key 'a' not found", json::out_of_range&);
+#endif
 
             json const doc3 = R"({ "a": {} })"_json;
             json const patch2 = R"([{ "op": "add", "path": "/a/b/c", "value": 1 }])"_json;
 
             // should cause an error because "b" does not exist in doc3
 #if JSON_DIAGNOSTICS
             CHECK_THROWS_WITH_AS(doc3.patch(patch2), "[json.exception.out_of_range.403] (/a) key 'b' not found", json::out_of_range&);
+#elif JSON_DIAGNOSTIC_POSITIONS
```

Comment:
Should these tests be run with both defined as well?

Reply:
@gregmarr I thought about that but I am not sure if there is ci_test target with both defined, @nlohmann please let me know what you think about this?

Reply:
In the CI, we have a dedicated test for `JSON_DIAGNOSTICS` and `JSON_DIAGNOSTIC_POSITIONS`. I don't think starting with combinations is helpful here. We maybe should rather have a dedicated test file where `JSON_DIAGNOSTICS` and `JSON_DIAGNOSTIC_POSITIONS` are defined together. This is then also automatically part of the coverage report.

Reply:
Normally I wouldn't think that having all combinations would be useful, but this is one case where the options interact.  A dedicated combined test might work just as well without increasing test run time too much.

Reply:
Yes, I am creating a new test case with both defined in a new test file similar to unit-diagnostics.cpp